### PR TITLE
add new package py-sonata-network-reduction

### PR DESCRIPTION
**Pull request by asanin-epfl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-bglibpy/package.py [hunk 1]
```diff
@@ -22,6 +23,6 @@ class PyBglibpy(PythonPackage):
     depends_on('py-bluepy@0.13.2:', type='run')
     depends_on('py-libsonata', type='run')
 
-    def setup_environment(self, spack_env, run_env):
+    def setup_run_environment(self, spack_env, run_env):
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-bglibpy/package.py]:
```suggestion
    def setup_run_environment(self, env):
```

File: var/spack/repos/builtin/packages/py-bglibpy/package.py [hunk 2]
```diff
@@ -22,6 +23,6 @@ class PyBglibpy(PythonPackage):
...
         run_env.set('NEURON_INIT_MPI', "0")
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-bglibpy/package.py]:
```suggestion
        env.set('NEURON_INIT_MPI', "0")
```

File: var/spack/repos/builtin/packages/py-bglibpy/package.py [hunk 3]
```diff
@@ -22,6 +23,6 @@ class PyBglibpy(PythonPackage):
...
         run_env.unset('PMI_RANK')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-bglibpy/package.py]:
```suggestion
        env.unset('PMI_RANK')
```

File: var/spack/repos/builtin/packages/py-sonata-network-reduction/package.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    depends_on('py-bglibpy@4.2:', type='run')
+    depends_on('py-morphio@2.3.4', type='run')
+    depends_on('neuron+python', type='run')
+    depends_on('py-aibs-circuit-converter@0.0.1', type='run')
+    depends_on('py-bluepysnap@0.1.2', type='run', when='@0.0.1')
+    depends_on('py-bluepysnap@0.2.0', type='run', when='@0.0.2:')
+    depends_on('py-neuron-reduce@0.0.7', type='run')
+
+
+    def setup_run_environment(self, spack_env, run_env):
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-sonata-network-reduction/package.py]:
```suggestion
    def setup_run_environment(self, env):
```

File: var/spack/repos/builtin/packages/py-sonata-network-reduction/package.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        run_env.set('NEURON_INIT_MPI', "0")
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-sonata-network-reduction/package.py]:
```suggestion
        env.set('NEURON_INIT_MPI', "0")
```

**Review by matz-e — CHANGES_REQUESTED** [review]

**Comment by asanin-epfl** [comment]
Now it's ready for the review. Should I consider codecov?

**Comment by matz-e** [comment]
> Now it's ready for the review. Should I consider codecov?

No.

**Review by matz-e — DISMISSED** [review]
I approve

**Comment by asanin-epfl** [comment]
Sorry, found this typo before pressing 'merge' button. Please review once more.

**Review by matz-e — APPROVED** [review]

**Pull request merged**